src/librustc/middle/stability.rs

```diff
@@ -685,15 +682,8 @@ fn is_internal<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, span: Span) -> bool {
 }
 
 fn is_staged_api<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, id: DefId) -> bool {
-    match tcx.trait_item_of_item(id) {
-        Some(trait_method_id) if trait_method_id != id => {
```

Comment:
Took me a bit to remember, but it never should've really mattered. Regardless of paat practices, we *never* refer to associated items in impls of traits, only to the definition in the trait, outside of monomorphization.